internal/translayer/dbuser/dbuser.go

```diff
@@ -0,0 +1,57 @@
+package dbuser
+
+import (
+	"context"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/translayer"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/controller/atlas"
+)
+
+type Service struct {
+	admin.DatabaseUsersApi
+}
+
+func NewService(ctx context.Context, provider atlas.Provider, secretRef *types.NamespacedName, log *zap.SugaredLogger) (*Service, error) {
+	client, err := translayer.NewVersionedClient(ctx, provider, secretRef, log)
+	if err != nil {
+		return nil, err
+	}
+	return &Service{DatabaseUsersApi: client.DatabaseUsersApi}, nil
+}
+
+func (dus *Service) Get(ctx context.Context, db, projectID, username string) (*User, error) {
+	atlasDBUser, _, err := dus.GetDatabaseUser(ctx, projectID, db, username).Execute()
+	if err != nil {
+		if admin.IsErrorCode(err, atlas.UsernameNotFound) {
+			return nil, nil
+		}
+		return nil, err
+	}
+	return toK8sDatabaseUser(atlasDBUser)
+}
+
+func (dus *Service) Delete(ctx context.Context, db, projectID, username string) (bool, error) {
+	_, _, err := dus.DeleteDatabaseUser(ctx, projectID, db, username).Execute()
+	if err != nil {
+		if admin.IsErrorCode(err, atlas.UsernameNotFound) {
+			return false, nil
```

Comment:
the consumer only knows that deletion failed (`false`) but not why.

Reply:
As said, we could translate the errors as well, the ones the consumer cares about.
In this case the bool mean `deleted` but in this case it was not, because it was not there already.

There are 2 ways to treat deletions:
- Non idempotent: producen an error if the item was already deleted.
- Idempotent: interpret the deletion command as successful is the item ceases to exists, so if it was deleted in this call or a subsequent one, it does not matter. Still you might want to distinguish if it was deleted now or already deleted.

I chose the idempotent option here, but we could revisit this decision and have a translated error case for that.